src/v0/destinations/sendgrid/deleteUsers.js

```diff
@@ -52,27 +76,27 @@ const userDeletionHandler = async (userAttributes, config) => {
       Authorization: `Bearer ${apiKey}`
     }
   };
-  const identity = [];
-  userAttributes.forEach(userAttribute => {
-    // Dropping the user if userId is not present
-    if (userAttribute.userId) {
-      identity.push(userAttribute.userId);
-    }
-  });
-
-  // batchEvents = [[e1,e2,e3,..batchSize],[e1,e2,e3,..batchSize]..]
-  // ref : https://developer.clevertap.com/docs/disassociate-api
-  const batchEvents = _.chunk(identity, MAX_BATCH_SIZE);
+  // batchEvents = [["e1,e2,e3,..urlLimit"],["e1,e2,e3,..urlLimit"]..]
+  // ref : https://docs.sendgrid.com/api-reference/contacts/delete-contacts
+  const batchEvents = chunksFromUrlLength(userAttributes, urlLimit);
```

Comment:
It would be better if we use lodash.chunk method here.

Reply:
The function is too much different from what we are trying to achieve from `chunksFromUrlLength ()` .